### ISSUE TITLE
Add Firefox support

### ISSUE THREAD

**Issue by ultranity** [tumuyan/Tree-Style-History#10]
希望能为Firefox提供支持，本身不是专业做前端的同时没做过浏览器拓展某些问题需要帮助

I found this ext very usful and I am trying to add support to firefox, it turns out that there just need a litte change in manifest to make it runable:
- firefox provide no support fof chrome://favicons so it needs to be removed, and therefore now the favicon of each history should leave empty (I am trying to find some api replacement but still no idea)
- firefox use options_ui instead of options_page
- firefox do not use update_url

there still have some bugs to fix:
- some chome-only url call using chrome_url  needs to be replaced by web extension format
- recent closed tab don't work (firefox use standard api browser.sessions.getRecentlyClosed instead of chrome.extension.getBackgroundPage().closedTabs)
- favicon

so if possible I hope there could be an offical support for firefox, which won't need too much time to make it work.

some detail can be found at https://www.extensiontest.com and https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API

**Comment by ultranity** [comment]
查阅FF文档发现FF可以兼容调用chrome.*形式的api，所以可能所需的工作量就更小了，期待ing

**Comment by tumuyan** [comment]
…实际上我也没不是这方面的专业人员，只是凑合着在别人的基础上增加了功能。
虽然现在大家都是WE扩展，Chrome API是部分兼容的，然而我之前已经简单试过，感觉不太行（所以目前内容放在了Chrome目录）

**Comment by yoghurtdewdew** [comment]
不好意思，没看懂大家的讨论，请教下大家，在 firefox 上是否可以使用这个插件呀？如何才能实现在 Firefox 上使用这个插件呢？

**Comment by tumuyan** [comment]
> 不好意思，没看懂大家的讨论，请教下大家，在 firefox 上是否可以使用这个插件呀？如何才能实现在 Firefox 上使用这个插件呢？

不止ff无法使用,甚至Chromium内核的浏览器也快不能用这个插件了…

**Comment by yoghurtdewdew** [comment]
> 不止ff无法使用,甚至Chromium内核的浏览器也快不能用这个插件了…

为什么💔 这个插件这么好用为啥不能😱

**Comment by Goosegit11** [comment]
I agree, this addon is something I've been looking for a long time, but I use Firefox.
